File: src/leetCodeProblems/Strings/CountOddNumbersInIntervalRange1523.java
package leetCodeProblems.Strings;

/**
 * LeetCode - https://leetcode.com/problems/count-odd-numbers-in-an-interval-range/submissions/
 *
 * TimeComplexity - O(1)
 */
public class CountOddNumbersInIntervalRange1523 {

    public int countOdds(int low, int high) {

        int ans = (high-low)/2;

        if (low%2 == 1 || high%2 == 1) {
            ans++;
        }

        return ans;
    }

    public static void main(String[] args) {

        CountOddNumbersInIntervalRange1523 obj = new CountOddNumbersInIntervalRange1523();

        System.out.println(obj.countOdds(3, 7));

        System.out.println(obj.countOdds(278382788, 569302584));
    }
}

File: src/leetCodeProblems/MathCalculations/CountOddNumbersInIntervalRange1523.go
package main

/**
 * LeetCode - https://leetcode.com/problems/count-odd-numbers-in-an-interval-range/
 * - Time - O(1)
 * - Space - O(1)
 */
func countOdds(low int, high int) int {

	out := (high - low) / 2

	if high%2 == 1 || low%2 == 1 {
		out++
	}

	return out
}

// func main() {
// 	log.Println(countOdds(3, 7))
// 	log.Println(countOdds(8, 10))
// }
